packages/models-library/src/models_library/api_schemas_webserver/functions_wb_schema.py

```diff
@@ -0,0 +1,230 @@
+from enum import Enum
+from typing import Annotated, Any, Literal, TypeAlias
+from uuid import UUID
+
+from models_library import projects
+from models_library.basic_regex import SIMPLE_VERSION_RE
+from models_library.services_regex import COMPUTATIONAL_SERVICE_KEY_RE
+from pydantic import BaseModel, Field, StringConstraints
+
+from ..projects import ProjectID
+
+FunctionID: TypeAlias = projects.ProjectID
+FunctionJobID: TypeAlias = projects.ProjectID
+FileID: TypeAlias = UUID
+
+InputTypes: TypeAlias = FileID | float | int | bool | str | list
+
+
+class FunctionSchema(BaseModel):
+    """Schema for function input/output"""
+
+    schema_dict: dict[str, Any] | None  # JSON Schema
+
+
+class FunctionInputSchema(FunctionSchema): ...
+
+
+class FunctionOutputSchema(FunctionSchema): ...
+
+
+class FunctionClass(str, Enum):
+    project = "project"
+    solver = "solver"
+    python_code = "python_code"
+
```

Comment:
here I think the same comments as for `functions.py`

Reply:
I deleted the functions.py and applied your comments here